### PR TITLE
feat: add `@joshdb/sqlite`

### LINKED ISSUE #32: feat: add `@joshdb/sqlite`

### PR DESCRIPTION
**Pull request by RealShadowNova**
Resolves #32 

- [x] Add tests for v1 migrations

### CODE REVIEW

File: .github/workflows/continuous-tests.yml [hunk 1]
```diff
@@ -124,10 +124,29 @@ jobs:
         with:
           name: coverage
           path: coverage/
+  sqlite:
+    name: RedisProvider
```
Comment (dan-online) [inline .github/workflows/continuous-tests.yml:128]: SQLite

File: packages/maria/src/lib/MariaProvider.ts [hunk 2]
```diff
@@ -723,9 +723,8 @@ export class MariaProvider<StoredValue = unknown> extends JoshProvider<StoredVal
   }
 
   public static defaultConnectionConfig: MariaProvider.ConnectionConfig = {
-    user: 'josh',
-    password: 'josh',
-    database: 'josh'
+    database: 'josh',
+    user: 'josh'
   };
```
Comment (dan-online) [inline packages/maria/src/lib/MariaProvider.ts:728]: Password is needed

File: packages/sqlite/package.json [hunk 3]
```diff
@@ -0,0 +1,61 @@
+{
+  "name": "@joshdb/sqlite",
+  "version": "2.0.0",
+  "description": "A Josh provider for @joshdb/core which uses Sqlite",
+  "author": "Évelyne Lachance <eslachance@gmail.com> (https://evie.codes/)",
+  "contributors": [],
```
Comment (dan-online) [inline packages/sqlite/package.json:6]: Add yourself here

**Review by dan-online — CHANGES_REQUESTED** [review]
I haven't read the SQLiteProvider file yet, but first these changes and also add the SQLite data file to gitignore

File: packages/sqlite/tests/lib/QueryHandler.test.ts [hunk 4]
```diff
@@ -0,0 +1,327 @@
+import Database from 'better-sqlite3';
+import { QueryHandler } from '../../src';
+
+describe('QueryHandler', () => {
+  describe('is a class', () => {
+    test('GIVEN typeof QueryHandler THEN returns function', () => {
```
[the remark this answers is not in the document]
Comment (RealShadowNova) [reply packages/sqlite/tests/lib/QueryHandler.test.ts:6]: These are just basically placeholders for the tests whenever they haven't been written yet. Never found a reason to remove them everywhere.

File: packages/sqlite/src/lib/SQLiteProvider.ts [hunk 5]
```diff
@@ -0,0 +1,846 @@
...
+        ? resolve(this.options.dataDirectory ?? 'data')
+        : resolve(process.cwd(), this.options.dataDirectory ?? 'data'),
+      wal: this.options.wal ?? true,
+      persistent: this.options.persistent ?? true,
+      disableSerialization: this.options.disableSerialization ?? false
+    };
+  }
+
+  public get version(): JoshProvider.Semver {
+    return process.env.NODE_ENV === 'test' ? { major: 2, minor: 0, patch: 0 } : this.resolveVersion('[VI]{version}[/VI]');
```
[the remark this answers is not in the document]
Comment (RealShadowNova) [reply packages/sqlite/src/lib/SQLiteProvider.ts:105]: Unless you can find a better way, this is the only way.

**Review by dan-online — APPROVED** [review]
LGTM

**Review by dan-online — APPROVED** [review]
LGTM

**Pull request merged**